stdlib/list.ml

```diff
@@ -459,7 +459,7 @@ let sort_uniq cmp l =
           let c = cmp x1 x2 in
           if c = 0 then
             let c = cmp x2 x3 in
-            if c = 0 then [x2] else if c < 0 then [x2; x3] else [x3; x2]
+            if c = 0 then [x1] else if c < 0 then [x2; x3] else [x3; x2]
```

Comment:
I have the impression that there is a mismatch between the code and the property you want. In this case we know that `cmp x1 x2 = 0`, and `x1` comes first in the list, so I would expect `x1` to be returned instead of `x2` in all cases. Why are `[x2; x3]` and `[x3; x2]` correct?